### ISSUE TITLE
[Feature Request] - Disable playback unless logged in

### ISSUE THREAD

**Issue by cpressland** [TeamPiped/Piped#3101]
### Describe the feature

We currently have the ability to restrict registration with `DISABLE_REGISTRATION`, it'd be nice to have a `DISABLE_ANONYMOUS_USAGE`/`REQUIRE_LOGIN` or similar option to prevent random internet users from using a public instance without being logged in.

### Why would this be useful to add?

My use case is a Piped instance that I plan to share with my friends which will be hosted out of my Homelab, while I do have a `/29` and a `/48`, I'm not sure if the `piped-proxy` is actively using the `/64` it's subnetted to effectively or just using whatever `/128` docker has NAT'd to it. 

Even if the proxy uses a `/32` or `/128`, a handful of users watching videos isn't going to get my home IP(s) blocked by Google, however, a swarm of random internet people will.

So I'd like the ability to lock down who can use the Piped Frontend/API/Proxy while I toggle registration on/off if I want to bring somebody else in. In the perfect world I could entirely restrict access to known IP addresses, and while some friends have static IPs, most do not.

### Acknowledgements

- [X] I have searched the existing issues and this is **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this form.

**Comment by FireMasterK** [comment]
Duplicate of #441

**Comment by cpressland** [comment]
Gah, sorry @FireMasterK!

**Comment by FireMasterK** [comment]
No worries :)
